launcher/sdw-notify.py

```diff
@@ -54,17 +54,44 @@ def show_update_warning():
     Show a graphical warning reminding the user to check for security updates
     using the preflight updater.
     """
+
     app = QApplication([])  # noqa: F841
 
-    QMessageBox.warning(
-        None,
-        "Security check recommended",
-        "This computer has not been checked for security updates recently. "
-        "We recommend that you launch or restart the SecureDrop app to "
-        "check for security updates.",
-        QMessageBox.Ok,
-        QMessageBox.Ok,
-    )
+    dialog = QMessageBox()
+    dialog.setWindowTitle(strings.headline_notify_updates)
+    dialog.setIcon(QMessageBox.Warning)
+    dialog.setStandardButtons(QMessageBox.No | QMessageBox.Ok)
+    dialog.setDefaultButton(QMessageBox.Ok)
+    button_check_now = dialog.button(QMessageBox.Ok)
+    button_check_now.setText(strings.button_check_for_updates)
+    button_defer = dialog.button(QMessageBox.No)
+    button_defer.setText(strings.button_defer_check)
+
+    if Util.is_sdapp_halted():
+        dialog.setText(strings.description_notify_updates)
+    else:
+        # If sd-app is running or state is unknown, use longer warning message
+        dialog.setText(strings.description_notify_updates_sdapp_running)
+
```

Comment:
I think this dialog could be extracted as a standalone component (widget).
The way I see it, it is a static dialog (the contents wouldn't change while the dialog is open) so the condition `Util.is_sdapp_halted()` could be a parameter to the constructor.

The API of the new class would be the same as the `QMessageBox` API, so none of the code below should need any change, that's what makes me think the refactoring could be a good idea.

Reply:
Naming-wise, my first thoughts would go towards "update prompt", "security check prompt"... what do you think @rocodes?